src/main/java/com/corndel/cashregister/Level5.java

```diff
@@ -28,7 +28,44 @@ public class Level5 {
    */
   public static List<Item> transaction(int cost, List<Item> paid, List<Item> drawer) {
     // TODO
-    return null;
+    /**
+     * Cost given and paid given
+     * calculate if change can be given
+     * if yes, add paid to drawer
+     * then remove change
+     * return drawer
+     */
+    System.out.println("1:" + drawer);
+    // Check what change is required and if it can be given
+    var changeRequired = Level3.sumDrawer(paid) - cost;
+    System.out.println(changeRequired);
+    boolean canGiveChange = Level4.canMakeAmount(changeRequired, drawer);
+    System.out.println("2:" + drawer);
+    // If change can be given, add payment into drawer
+    if (canGiveChange){
+      for (var item : paid){
+        Level1.addItem(item.name, drawer);
+        System.out.println(item.name + ":" + item.quantity);
+
+      }
+      System.out.println(drawer);
+
+      // Then make change.
+      for (var item : drawer) {
```

Comment:
This block of code is duplicated. The `Level4.canMakeAmount(changeRequired, drawer)` method will update the items in the drawer.

If you look at your print statements, you will see that it is removing twice the amount needed.

If you remove this for loop, the test for $20 will pass. 👍